deployment/tfhub/object-detection-url.ipynb

```diff
@@ -0,0 +1,643 @@
+        "    def load_img(self, path):\n",
+        "        img = tf.io.read_file(path)\n",
+        "        img = tf.image.decode_jpeg(img, channels=3)\n",
+        "        \n",
+        "        return img\n",
+        "\n",
+        "    def filter_results(self, file, response, entity='Car', min_score=.2):\n",
+        "        unused_keys = ['detection_class_labels', 'detection_class_names']\n",
+        "        response = {key: value.numpy().tolist() for key, value in response.items()}\n",
+        "        response = {key: val for key, val in response.items() if key not in unused_keys}\n",
+        "        response['detection_class_entities'] = [v.decode() for v in response['detection_class_entities']]\n",
+        "\n",
+        "        entities = response['detection_class_entities']\n",
+        "        scores = response['detection_scores']\n",
+        "        bboxes = response['detection_boxes']\n",
+        "        result = {}\n",
+        "\n",
+        "        for i in range(len(entities)):\n",
+        "            if entities[i] == entity and scores[i] >= min_score:\n",
+        "                ymin, xmin, ymax, xmax = bboxes[i]\n",
+        "                result = {\n",
+        "                    'file': file,\n",
+        "                    'has_car': True,\n",
+        "                    'score': scores[i],\n",
+        "                    'bboxes': {'ymin': ymin, 'xmin': xmin, 'ymax': ymax, 'xmax': xmax}\n",
+        "                }\n",
+        "                break\n",
+        "\n",
+        "        if len(result) == 0:\n",
+        "            result = {\n",
+        "                'file': file,\n",
+        "                'has_car': False,\n",
+        "                'score': 0,\n",
+        "                'bboxes': {'ymin': 0, 'xmin': 0, 'ymax': 0, 'xmax': 0}\n",
+        "            }\n",
+        "        \n",
+        "        print(f\"Found {result['has_car']} object(s).\")\n",
+        "        return result\n",
+        "\n",
+        "    def run_detector(self, file, image_path):\n",
+        "        img = self.load_img(image_path)\n",
+        "        img = tf.image.convert_image_dtype(img, tf.float32)[tf.newaxis, ...]\n",
+        "        \n",
+        "        response = self.detector(img)\n",
+        "        result = self.filter_results(file, response)\n",
+        "        \n",
+        "        return result\n",
+        "\n",
+        "    def detect_objects(self, data):\n",
+        "        start_time = time.time()\n",
+        "        image_path = self.handle_img(data)\n",
+        "        result = self.run_detector(data[0], image_path)\n",
+        "        end_time = time.time()\n",
+        "\n",
+        "        print(f\"Inference time: {end_time - start_time}.\")\n",
+        "        return result\n",
+        "\n",
+        "    @verify_io\n",
+        "    def predict(self, data):\n",
+        "        result = self.detect_objects(data)\n",
```

Comment:
I would recommend splitting data as

```python
file_name, url  = data
```

similar to what we do in the other notebook. Otherwise, seeing `data[0]` and `data[1]` in the other functions can be a little confusing and unclear.

Reply:
Done.